kernel/fork.c

```diff
@@ -172,6 +173,13 @@ dword_t sys_clone(dword_t flags, addr_t stack, addr_t ptid, addr_t tls, addr_t c
 
     // task might be destroyed by the time we finish, so save the pid
     pid_t pid = task->pid;
+
+    if (current->traced) {
+        current->ptrace.trap_event = PTRACE_EVENT_FORK_;
+        lock(&pids_lock);
```

Comment:
Missing unlock(&pids_lock), but I dunno if the lock is necessary (see other comments)